ztest/shell.go

```diff
@@ -2,19 +2,28 @@ package ztest
 
 import (
 	"bytes"
+	"fmt"
 	"io"
+	"os"
 	"os/exec"
 	"runtime"
 )
 
-func RunShell(dir *Dir, bindir, script string, stdin io.Reader) (string, string, error) {
+func RunShell(dir *Dir, bindir, script string, stdin io.Reader, useenvs []string) (string, string, error) {
 	var cmd *exec.Cmd
 	if runtime.GOOS == "windows" {
 		cmd = exec.Command("cmd.exe", "/c", script)
 	} else {
 		cmd = exec.Command("bash", "-c", script)
 	}
-	cmd.Env = []string{"PATH=/bin:/usr/bin:" + bindir}
+
+	for _, env := range useenvs {
+		if v := os.Getenv(env); v != "" {
+			cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", env, v))
+		}
```

Comment:
Don't drop variables with empty values.
```suggestion
		if v, ok := os.LookupEnv(env); ok {
			cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", env, v))
		}
```